internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -481,38 +475,37 @@ func (r *PreprovisioningImageReconciler) getIronicDefaultConfig(log logrus.Field
 	}
 
 	log.Infof("Setting default ironic agent image (%s)", ironicAgentImage)
-	return &ICCConfig{
-		IronicAgentImage: ironicAgentImage,
-	}
+	return ironicAgentImage
 }
 
 func (r *PreprovisioningImageReconciler) getIronicConfig(ctx context.Context, log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv, infraEnvInternal *common.InfraEnv) (*ICCConfig, error) {
-	var iccConfig *ICCConfig
+	iccConfig := r.getIronicConfigFromBMOConfig(ctx, log, infraEnvInternal)
+	if iccConfig == nil {
+		iccConfig = &ICCConfig{}
+	}
 
-	iccConfig = r.getIronicConfigFromUserOverride(log, infraEnv)
+	if iccConfig.IronicBaseURL == "" || iccConfig.IronicInspectorBaseUrl == "" {
```

Comment:
No, this needs to be an `&&` not `||` alternatively you could check for nil here  like this:

```go
iccConfig := r.getIronicConfigFromBMOConfig(ctx, log, infraEnvInternal)
if iccConfig == nil {
  iccConfig = &ICCConfig{}
  if err := r.fillIronicServiceURLs(ctx, infraEnv, infraEnvInternal, iccConfig); err != nil {
    return nil, err
  }
}
```

I think this shows more clearly that we're getting the info from the cluster only of we can't find the configmap.